tests/integration/hosts/maya/input/startup/userSetup.py

```diff
@@ -0,0 +1,26 @@
+import logging
+import sys
+
+from maya import cmds
+
+
+def setup_pyblish_logging():
+    log = logging.getLogger("pyblish")
+    hnd = logging.StreamHandler(sys.stdout)
+    fmt = logging.Formatter(
+        "pyblish (%(levelname)s) (line: %(lineno)d) %(name)s:"
+        "\n%(message)s"
+    )
+    hnd.setFormatter(fmt)
+    log.addHandler(hnd)
+
+
+def main():
+    cmds.evalDeferred("setup_pyblish_logging()", evaluateNext=True)
+    cmds.evalDeferred(
+        "import pyblish.util;pyblish.util.publish()", lowestPriority=True
+    )
+    cmds.evalDeferred("cmds.quit(force=True)", lowestPriority=True)
```

Comment:
If anything calls `cmds.refresh(force=True)` (and maybe even without `force`) doesn't this quit already trigger since it's forced to propagate?

Doesn't this make much more sense to be explicit part of the Maya tests "teardown" code to force it's only run once teardown triggers?

Reply:
Not entirely sure how you envisage this?

The current idea with the tests, is to focus on publishing successfully so that means launching the app, running a publish and inspect the results. I guess we could register a plugin to run at the very end of the publishing to quit Maya?

Reply:
> I guess we could register a plugin to run at the very end of the publishing to quit Maya?

That does already sound more trust-worthy, but having something that explicitly encapsulates the full run by design, like a test runner usually does, would be less ambiguous and better design I think.

Reply:
If we are too lazy to refactor, then at least make sure we ourselves control the order instead of executing each individually:
```python
import logging
import sys

import pyblish.util
    
from maya import cmds


def setup_pyblish_logging():
    log = logging.getLogger("pyblish")
    hnd = logging.StreamHandler(sys.stdout)
    fmt = logging.Formatter(
        "pyblish (%(levelname)s) (line: %(lineno)d) %(name)s:"
        "\n%(message)s"
    )
    hnd.setFormatter(fmt)
    log.addHandler(hnd)
    
    
def _run_publish_test_deferred():
    setup_pyblish_logging()
    pyblish.util.publish()
    cmds.quit(force=True)

def main():
    cmds.evalDeferred("_run_publish_test_deferred()", lowestPriority=True)

main()
```

If `pyblish.util.publish()` raises error that we do NOT want to interrupt the quitting of maya itself then we should do a:
```python
try:
    # whatever we want to do
    pyblish.util.publish()
finally:
    # quit maya
    cmds.quit(force=True)
```

Reply:
That might be good solution!